File: examples/governance_static_go/domain/finding.go
package domain

type Finding struct {
	Category string `json:"category"`
	Severity string `json:"severity"`
	Description string `json:"description"`
}

func NewFinding(category string, severity string, description string) (Finding, error) {
	v := Finding{
		Category: category,
		Severity: severity,
		Description: description,
	}
	return v, nil
}

File: examples/governance/risk_assessment_domain/lib/risk_assessment_domain/assessment/finding.rb
require 'hecks/mixins/model'

module RiskAssessmentDomain
  class Assessment
    class Finding
      include Hecks::Model

      attribute :category
      attribute :severity
      attribute :description
      attribute :status

      private

      def check_invariants!
        raise RiskAssessmentDomain::InvariantError, "severity must be valid" unless instance_eval(&proc {  })
      end
    end
  end
end
